src/components/organisms/AssetActions/Compute/index.tsx

```diff
@@ -0,0 +1,300 @@
+
+      !hasPreviousDatasetOrder &&
+        !hasDatatoken &&
+        (await buyDT('1', (ddo as DDO).price, ddo))
+
+      !hasPreviousAlgorithmOrder &&
+        !hasAlgoAssetDatatoken &&
+        (await buyDT(
+          '1',
+          (selectedAlgorithmAsset as DDO).price,
+          selectedAlgorithmAsset
+        ))
+
+      const allowed = await ocean.compute.isOrderable(
+        ddo.id,
+        computeService.index
+      )
+
+      console.log('is dataset orderable ?', allowed)
+      const assetOrderId = hasPreviousDatasetOrder
+        ? previousDatasetOrderId
+        : await ocean.compute.orderAsset(
+            accountId,
+            ddo.id,
+            computeService.index
+          )
+
+      const algorithmAssetOrderId = hasPreviousAlgorithmOrder
+        ? previousAlgorithmOrderId
+        : await ocean.compute.orderAlgorithm(
+            algorithmId,
+            serviceAlgo.type,
+            accountId,
+            serviceAlgo.index
+          )
+
+      const output = {}
+      const respone = await ocean.compute.start(
+        ddo.id,
+        assetOrderId,
+        ddo.dataToken,
+        account,
+        algorithmId,
+        undefined,
+        output,
+        `${computeService.index}`,
+        computeService.type,
+        algorithmAssetOrderId,
+        selectedAlgorithmAsset.dataToken
+      )
+      console.log('respone', respone)
+
+      setHasPreviousDatasetOrder(true)
+      setIsPublished(true)
+    } catch (error) {
+      console.log('error found')
+      setError('Failed to start job!')
+      Logger.error(error.message)
+    } finally {
+      setIsJobStarting(false)
+    }
+  }
+
+  async function handleSubmit(
+    values: any,
+    resetForm: () => void
+  ): Promise<void> {
+    await startJob(values.algorithm)
+  }
+
+  return (
+    <>
+      <div className={styles.info}>
+        <File file={file} small />
+        <Price price={(ddo as DDO).price} conversion />
+      </div>
+
+      {type === 'algorithm' ? (
+        <Input
+          type="select"
+          name="data"
+          label="Select dataset for the algorithm"
+          placeholder=""
+          size="small"
+          value="dataset-1"
+          options={['dataset-1', 'dataset-2', 'dataset-3'].map((x) => x)}
+          // onChange={handleSelectChange}
+        />
+      ) : (
+        <Formik
+          initialValues={getInitialValues}
+          validationSchema={validationSchema}
+          onSubmit={async (values, { resetForm }) => {
+            await handleSubmit(values, resetForm)
+          }}
+        >
+          <FormStartComputeDataset
+            algorithms={algorithmList}
+            ddoListAlgorithms={ddoAlgorithmList}
+            setSelectedAlgorithm={setSelectedAlgorithmAsset}
+            isLoading={isJobStarting}
+            isComputeButtonDisabled={isComputeButtonDisabled}
+            hasPreviousOrder={
+              hasPreviousDatasetOrder || hasPreviousAlgorithmOrder
```

Comment:
this most likely will have to be tweaked cause now we have to consider 2 different previous orders. Which sounds like we should update the messaging accordingly in `ButtonBuy`, and then tweak what we pass here